packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -193,14 +208,128 @@ function CollabSidebarContent( {
 	);
 }
 
+function CommentBoardWrapper( {
+	thread,
+	showCommentBoard,
+	setShowCommentBoard,
+	backgroundColor,
+	previousThreadId,
+	commentSidebarRef,
+	offsetsRef,
+	updateOffsets,
+	updateHeight,
+	heights,
+} ) {
+	const blockRef = useRef();
+	useBlockElementRef( thread.blockClientId, blockRef );
+
+	const selectedBlockElementRect = blockRef.current?.getBoundingClientRect();
+
+	const initialOffsetTop = selectedBlockElementRect?.top;
+
+	const previousOffset = previousThreadId
+		? offsetsRef.current[ previousThreadId ]
+		: 0;
+
+	const previousBoardHeight = heights[ previousThreadId ]
+		? heights[ previousThreadId ]
+		: 0;
+
+	// If the previous comment board is overlapping this comment, shift it down.
+	const calculateOffset = () => {
+		if (
+			previousOffset &&
+			initialOffsetTop < previousOffset + previousBoardHeight
+		) {
+			return previousOffset - initialOffsetTop + previousBoardHeight + 20;
+		}
+		return -16; // Remove top padding of the comment board so first comment visually aligns with block.
+	};
+
+	// Use floating-ui to track the block element's position. The crossAxis offset
+	// is calculated to avoid overlapping comment boards and will shift the board down.
+	const { y, refs } = useFloating( {
+		placement: 'right-start',
+		middleware: [
+			offsetMiddleware( {
+				crossAxis: calculateOffset(),
+			} ),
+		],
+		whileElementsMounted: autoUpdate,
+	} );
+
+	useEffect( () => {
+		if ( blockRef.current ) {
+			refs.setReference( blockRef.current );
+		}
+	}, [ blockRef, refs ] );
+
+	useEffect( () => {
+		if ( y !== null && y !== 0 ) {
+			updateOffsets( thread.id, y, refs.floating?.current?.clientHeight );
+		}
+	}, [ y, refs.floating, thread.id ] );
+
+	useEffect( () => {
+		if ( refs.floating?.current ) {
+			const newHeight = refs.floating?.current.scrollHeight;
+			updateHeight( thread.id, newHeight );
+		}
+	}, [ thread.id, updateHeight ] );
+
+	return (
+		<VStack
+			ref={ refs.setFloating }
+			className={ clsx( 'editor-collab-sidebar-panel__thread', {
+				'editor-collab-sidebar-panel__active-thread': false,
+			} ) }
```

Comment:
```suggestion
			className="editor-collab-sidebar-panel__thread is-floating"
```

The `,editor-collab-sidebar-panel__active-thread` selector doesn't exist. Instead, we can apply special CSS for floating.